hack/build_multiarch_clis.sh

```diff
@@ -9,58 +9,148 @@ set -euo pipefail
+  local GH_OUT="${GH_ARCHIVES}/flightctl-${GOOS}-${GOARCH}${TGZ}"
+
+  mkdir -p "${BIN}" "${ARCHIVES}" "${GH_ARCHIVES}"
+
+  cp "bin/flightctl${EXE}" "${BIN}/"
+  cp "bin/flightctl${EXE}" "flightctl-${GOOS}-${GOARCH}${EXE}"
 
+  if [ "${GOOS}" == "linux" ]; then
+    tar -zhcf "${ARCHIVES}/flightctl.tar.gz" -C "${BIN}" flightctl
+  else
+    zip -9 -r -q -j "${ARCHIVES}/flightctl.zip" "${BIN}/flightctl${EXE}"
+  fi
+
+  cp "${ARCHIVES}/flightctl${TGZ}" "${GH_OUT}"
+  sha256sum "${GH_OUT}" | awk '{ print $1 }' > "${GH_OUT}.sha256"
+}
 
 for GOARCH in amd64 arm64; do
   for GOOS in linux darwin windows; do
-    echo -e "\033[0;37m>>>> building cli for GOARCH=${GOARCH} GOOS=${GOOS}\033[0m"
-    GOARCH="${GOARCH}" GOOS="${GOOS}" make build-cli
-    OS="${GOOS}"
-    TGZ=".tar.gz"
-    EXE=""
-    if [ "${GOOS}" == "darwin" ]; then
-      OS="mac"
-      TGZ=".zip"
-    elif [ "${GOOS}" == "windows" ]; then
-      TGZ=".zip"
-      EXE=".exe"
-    fi
-    BIN="bin/clis/binaries/${GOARCH}/${OS}"
-    ARCHIVES="bin/clis/archives/${GOARCH}/${OS}"
-    GH_ARCHIVES="bin/clis/gh-archives/${GOARCH}/${OS}"
-    GH_OUT="${GH_ARCHIVES}/flightctl-${GOOS}-${GOARCH}${TGZ}"
-
-    mkdir -p "${BIN}"
-    mkdir -p "${ARCHIVES}"
-    mkdir -p "${GH_ARCHIVES}"
-
-    cp "bin/flightctl${EXE}" "${BIN}/"
-    cp "bin/flightctl${EXE}" "flightctl-${GOOS}-${GOARCH}${EXE}"
-
-    if [ "${GOOS}" == "linux" ]; then
-      tar -zhcf "${ARCHIVES}/flightctl.tar.gz" -C "${BIN}" flightctl
-    else
-      zip -9 -r -q -j "${ARCHIVES}/flightctl.zip" "${BIN}/flightctl${EXE}"
-    fi
-    cp "${ARCHIVES}/flightctl${TGZ}" "${GH_OUT}"
-    sha256sum "${GH_OUT}" | awk '{ print $1 }' > "${GH_OUT}.sha256"
+    build "$GOARCH" "$GOOS" &
   done
 done
 
+# Wait for all parallel jobs to complete
+wait
+
+echo -e "\033[0;37m>>>> building index.json\033[0m"
+
+SOURCE_DIR="bin/clis/gh-archives"
+OUTPUT_JSON="index.json"
+PLACEHOLDER_BASE_URL="{{CLI_ARTIFACTS_BASE_URL}}"
+
+cd "$SOURCE_DIR"
+
+# Build JSON array of artifact entries
+artifacts=$(find . -type f | while read -r file; do
+  filename=$(basename "$file")
+
+  # Skip index.json and .sha256 files
+  [[ "$filename" == "$OUTPUT_JSON" || "$filename" == *.sha256 ]] && continue
+
+  # Extract arch and os from path
+  parts=(${file//\// })
+  arch="${parts[1]}"
+  os="${parts[2]}"
+
+  # Read SHA256 if it exists
+  sha256_file="${file}.sha256"
+  sha256_content=""
+  [[ -f "$sha256_file" ]] && sha256_content=$(cat "$sha256_file")
+
+  # Emit artifact JSON object
+  jq -n \
+    --arg os "$os" \
+    --arg arch "$arch" \
+    --arg filename "$filename" \
+    --arg sha256 "$sha256_content" \
+    '{
+      os: $os,
+      arch: $arch,
+      filename: $filename,
+      sha256: $sha256
+    }'
+done | jq -s '.')
+
+# Write final JSON with baseUrl as placeholder
+jq -n \
+  --arg baseUrl "$PLACEHOLDER_BASE_URL" \
+  --argjson artifacts "$artifacts" \
+  '{baseUrl: $baseUrl, artifacts: $artifacts}' > "$OUTPUT_JSON"
```

Comment:
the baseUrl is known only in runtime, not in build time. So the replacement needs to happen via nginx (ie using `sub_filter` or similar)

Reply:
done